### PR TITLE
Create molecule core

### PR DESCRIPTION
**Pull request by johnduarte**
This PR creates a central gathering place for rpc-openstack-system-tests based on the [molecule framework](https://molecule.readthedocs.io/en/latest/).

* A `molecules` directory is created for molecule git submodules to be added. An initial module is included. 
* Python virtualenv requirements and constraints are added.
* A moleculerize.py script is added to transpose openstack-ansible the dynamic inventory into the molecule configuration file format.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "molecules/molecule-validate-cinder-deploy"]
+	path = molecules/molecule-validate-cinder-deploy
+	url = https://github.com/phongdly/molecule-validate-cinder-deploy
```
Comment (hughsaunders) [inline .gitmodules]: We should probably move that into rcbops to aid collaboration.

Comment (odyssey4me) [reply .gitmodules]: Agreed.

Comment (johnduarte) [reply .gitmodules]: They may be collapsed in the future, but I wanted to provide a level of indirection while we sort out exactly how the molecule tests should be composed.
@phongdly can you move your repo the rcbops space.

Comment (johnduarte) [reply .gitmodules]: @ryan-rs could you add a couple of tests for moleculerize.py ?

Comment (ryan-rs) [reply .gitmodules]: @johnduarte Sure. I'll hop on it after lunch.

**Comment by johnduarte** [comment]
@hughsaunders @odyssey4me I have added a local script to run the molecule tests and tarball the JUnit XML results.
@phongdly has moved the submodule for the test to the `rcbops` org.
@ryan-rs is working on tests for the moleculerize.py, we are hoping these are not a blocker merging this for the prototype.

Here is a screen session of of the process being kicked off by the `rpc-openstack-system-tests/run_system_tests.sh` script at https://github.com/rcbops/rpc-openstack/pull/2860/commits/0e1dd7dbe73c3087ad2b823b662e7cb2dbb36a22, using this repo at https://github.com/rcbops/rpc-openstack-system-tests/pull/2/commits/e0674cc1161c0325f7efbcef1ee46439bcb64b37

[rpc-openstack-system-tests.run.log](https://github.com/rcbops/rpc-openstack-system-tests/files/1753505/rpc-openstack-system-tests.run.log)

**Comment by johnduarte** [comment]
@hughsaunders The unit tests for `moleculerize.py` have been added by @ryan-rs . Let us know if there are any other blockers to merging this PR. Cheers.

**Comment by johnduarte** [comment]
@hughsaunders are there any other changes needed to make to get your approval to merge this PR?

File: execute_tests.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    ${MNAIO_SSH} "${SYS_INVENTORY}/dynamic_inventory.py" > dynamic_inventory.json
+else
+    ${SYS_INVENTORY}/dynamic_inventory.py > dynamic_inventory.json
+fi
+
+# Run molecule converge and verify
+# for each submodule in ${SYS_TEST_SOURCE}/molecules
+set +e # allow test stages to return errors
+for TEST in molecules/* ; do
+    ./moleculerize.py --output "$TEST/molecule/default/molecule.yml" /tmp/dynamic_inventory.json
```
Comment (ryan-rs) [inline execute_tests.sh]: @johnduarte This always reads from `/tmp/dynamic_inventory.json`, but it appears to drop the `dynamic_inventory.json` in the current working directory.

**Comment by phongdly** [comment]
LGTM and all contributors approved it. I am going to merge this PR

**Pull request merged**